### PR TITLE
[MIG] sale_order_import: Migration to 14.0

### PR DESCRIPTION
**Pull request by phuctranfxvn**
Requires modules:
- `sale_commercial_partner` (https://github.com/OCA/sale-workflow/tree/14.0)
- `base_business_document_import migrated` in https://github.com/OCA/edi/pull/301
- `onchange_helper` (https://github.com/OCA/server-tools/tree/14.0)

### CODE REVIEW

**Comment by etobella** [comment]
@phuctranerp 
I made a small Proof Of Concept about my comments about integrating this to `edi`. As `edi` has not been migrated, I had to made it on 13, but it should be easily migrated.

https://github.com/tegin/edi/tree/13.0-add-edi_sale/edi_sale

That is the idea. It maintains the same logic, but using `edi`. Some changes should be needed on `base_business_document_import` and move the functions to a component, but that should be easy. WDYT?
It was done on:

https://github.com/tegin/edi/tree/13.0-mig-account_invoice_import/base_business_document_import

But I didn't want to inherit in two many branches this change :wink:

**Comment by etobella** [comment]
Obviously, some refactoring is needed in order to not loose all the existing commits and contributors

**Review by fshah-initos — APPROVED** [review]
LGTM. Code review and Functional Tests.

**Comment by fshah-initos** [comment]
Hi @phuctranerp can you please resolve the conflicts?

**Comment by phuctranfxvn** [comment]
> Hi @phuctranerp can you please resolve the conflicts?

I have rebased the code with branch 14.0 and resolve the conflicts. Thanks

**Comment by fshah-initos** [comment]
It would be great if some could add review here so that this PR can be merged.

**Comment by fshah-initos** [comment]
It would be great if some could add review here so that this PR can be merged.

**Comment by fshah-initos** [comment]
It would be great if some could add review here so that this PR can be merged.

File: sale_order_import/readme/CONTRIBUTORS.rst [hunk 1]
```diff
@@ -1,3 +1,5 @@
 * Alexis de Lattre <alexis.delattre@akretion.com>
 * Dennis Sluijk <d.sluijk@onestein.nl>
 * Andrea Stirpe <a.stirpe@onestein.nl>
+* Tam (Nguyen Duc) <tamnd@trobz.com>
```
Comment (simahawk) [inline sale_order_import/readme/CONTRIBUTORS.rst]: @phuctranerp sorry, bare migration does not entitle as contributors. To land in this list you should at least add something (eg: fix a bug, test coverage, new features, significant refactoring, etc).
Can you pls apply the same concept to all your PRs?

File: sale_order_import/readme/CREDITS.rst [hunk 2]
```diff
@@ -0,0 +1,3 @@
+The development of this module has been financially supported by:
```
Comment (simahawk) [inline sale_order_import/readme/CREDITS.rst]: this is not required :wink:

**Review by simahawk — CHANGES_REQUESTED** [review]

File: sale_order_import/readme/CREDITS.rst (hunk 2 above)
Comment (nilshamerlinck) [reply sale_order_import/readme/CREDITS.rst]: This was a request from @chroi :-)
That said, we need to improve the message to be more specific: `The migration of this module from 13.0 to 14.0 was financially supported by Camptocamp`

File: sale_order_import/readme/CONTRIBUTORS.rst (hunk 1 above)
Comment (phuctranfxvn) [reply sale_order_import/readme/CONTRIBUTORS.rst]: Hi, thanks for your suggestion. I'll update it soon.

Comment (phuctranfxvn) [reply sale_order_import/readme/CONTRIBUTORS.rst]: updated

**Comment by fshah-initos** [comment]
Is it possible to merge this PR?

File: sale_order_import/readme/CREDITS.rst (hunk 2 above)
Comment (phuctranfxvn) [reply sale_order_import/readme/CREDITS.rst]: Hi, fyi, updated as `The migration of this module from 13.0 to 14.0 was financially supported by Camptocamp`

**Comment by fshah-initos** [comment]
It would be great if we could merge this PR

**Comment by fshah-initos** [comment]
@OCA/edi-maintainers can you please merge this PR?

**Review by simahawk — APPROVED** [review]

**Pull request merged**